runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -4185,6 +4185,37 @@ class INTERPRETER_CLASS
 
 		return rc;
 	}
+
+	VMINLINE VM_BytecodeAction
+	inlUnsafeIsFieldAtOffsetFlattened(REGISTER_ARGS_LIST)
+	{
+		I_64 offset = *(I_64*)_sp;
+		j9object_t clz = *(j9object_t*)(_sp + 2);
+
+		I_32 isFlattened = (I_32)FALSE;
+
+		if (NULL != clz) {
+			J9Class *clzJ9Class = J9VM_J9CLASS_FROM_HEAPCLASS(_currentThread, clz);
+			if ((!J9CLASS_IS_ARRAY(clzJ9Class)) && (NULL != clzJ9Class->flattenedClassCache)) {
+
+				/* caller provided offset will include header size but entry->offset does not */
+				offset -= (I_64)J9VMTHREAD_OBJECT_HEADER_SIZE(_currentThread);
+
+				UDATA numberOfFlattenedFields = clzJ9Class->flattenedClassCache->numberOfEntries;
+				for (UDATA i = 0; i < numberOfFlattenedFields; i++) {
+					J9FlattenedClassCacheEntry *entry = J9_VM_FCC_ENTRY_FROM_CLASS(clzJ9Class, i);
+					if ((I_64)entry->offset == offset) {
```

Comment:
Just talked with @ehrenjulzert , an additional check of `J9_IS_FIELD_FLATTENED(class, entry->field) `is needed.

Reply:
updated in ba9d334